tests/api/test_starship_engine.py

```diff
@@ -0,0 +1,97 @@
+# Copyright 2020 Xanadu Quantum Technologies Inc.
+
+# Licensed under the Apache License, Version 2.0 (the "License");
+# you may not use this file except in compliance with the License.
+# You may obtain a copy of the License at
+
+#     http://www.apache.org/licenses/LICENSE-2.0
+
+# Unless required by applicable law or agreed to in writing, software
+# distributed under the License is distributed on an "AS IS" BASIS,
+# WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+# See the License for the specific language governing permissions and
+# limitations under the License.
+"""TODO"""
+import numpy as np
+import pytest
+
+from strawberryfields.api import Connection, Job, JobStatus, Result
+from strawberryfields.engine import StarshipEngine
+
+
+def mock_return(return_value):
+    """A helper function for defining a mock function that returns the given value for
+    any arguments.
+    """
+    return lambda *args, **kwargs: return_value
+
+
+class MockServer:
+    """A mock platform server that fakes a processing delay by counting requests."""
+
+    REQUESTS_BEFORE_COMPLETED = 3
+
+    def __init__(self):
+        self.request_count = 0
+
+    def get_job_status(self, _id):
+        """Returns a 'queued' job status until the number of requests exceeds a defined
+        threshold, beyond which a 'complete' job status is returned.
+        """
+        self.request_count += 1
+        return (
+            JobStatus.COMPLETED
+            if self.request_count >= self.REQUESTS_BEFORE_COMPLETED
+            else JobStatus.QUEUED
+        )
+
+
+@pytest.fixture
+def job_to_complete(connection, monkeypatch):
+    """Mocks a remote job that is completed after a certain number of requests."""
+    monkeypatch.setattr(
+        Connection,
+        "create_job",
+        mock_return(Job(id_="123", status=JobStatus.OPEN, connection=connection)),
+    )
+    server = MockServer()
+    monkeypatch.setattr(Connection, "get_job_status", server.get_job_status)
```

Comment:
I agree that it looks a little jarring, but the files in this PR are all auto-formatted with `black` (which has a default line length limit of 88 characters), so I'm hesitant to manually impose ad hoc formatting :slightly_smiling_face:

Reply:
@pauljxtan, we actually use a line length of 100 characters on SF! So you might need to reblack before merging (`black -l 100 strawberryfields/`).

Before the next release, I am going to add a CI check that ensures blacking is done with the right line length, sorry about that!

Reply:
Ah I see, I will reformat the relevant files - thanks for pointing that out!

Reply:
@josh146 - running `black -l 100` it on the full `strawberryfields/` seemed to change a lot of unrelated modules, so I've run it only on the files relevant to this PR.

Reply:
good idea @pauljxtan

Reply:
:+1: Just prior to the new SF release, I plan on blacking the entire library, and adding a formatting CI check. This will stop new PRs from allowing unformatted code through.